src/Symfony/Bundle/FrameworkBundle/Controller/Controller.php

```diff
@@ -64,8 +64,10 @@ protected function generateUrl($route, $parameters = array(), $referenceType = U
      */
     protected function forward($controller, array $path = array(), array $query = array())
     {
+        $request = $this->container->get('request_stack')->getCurrentRequest();
+        $path['_forwarded'] = $request->attributes;
```

Comment:
I think this should be moved to an event listener instead. Otherwise you couldn't track subrequests triggered by user code that doesn't use the base controller class.